app/components/Admin/Popups/AddDJsPopup.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useState, useContext } from "react";
+
+import {
+    Box,
+    Button,
+    Checkbox,
+    CircularProgress,
+    FormControl,
+    FormLabel,
+    IconButton,
+    Input,
+    Sheet,
+    Stack,
+    Tab,
+    TabList,
+    TabPanel,
+    Table,
+    Tabs,
+    Textarea,
+    Tooltip,
+    Typography,
+} from "@mui/joy";
+import { toast } from "sonner";
+import { addDJ, applicationSlice, fetchDJs, populateStationManagers, populateMusicDirectors, useDispatch, AdminType } from "@/lib/redux";
+
+type AddDJsPopupProps = { };
+
+/**
+ * Represents a popup component for adding DJs.
+ * Only accessible to station managers.
+ *
+ * @component
+ * @category Station Management
+ *
+ * @param {Object} props - The component props.
+ * @param {function} props.callback - The callback function to be executed after adding DJs.
+ * @param {string} [props.style] - The style of the popup. Defaults to the success color.
+ *
+ * @returns {JSX.Element} The AddDJsPopup component.
+ */
```

Comment:
This documentation is great. Could you please add docs in other places? A lot of the code I'm left reverse engineering what the intent is.